server/src/test/java/org/elasticsearch/threadpool/ThreadPoolStatsTests.java

```diff
@@ -18,27 +18,63 @@
+import static org.hamcrest.Matchers.hasSize;
 
 public class ThreadPoolStatsTests extends ESTestCase {
     public void testThreadPoolStatsSort() {
-        var stats = List.of(
-            new ThreadPoolStats.Stats("z", -1, 0, 0, 0, 0, 0L),
-            new ThreadPoolStats.Stats("m", 3, 0, 0, 0, 0, 0L),
-            new ThreadPoolStats.Stats("m", 1, 0, 0, 0, 0, 0L),
-            new ThreadPoolStats.Stats("d", -1, 0, 0, 0, 0, 0L),
-            new ThreadPoolStats.Stats("m", 2, 0, 0, 0, 0, 0L),
-            new ThreadPoolStats.Stats("t", -1, 0, 0, 0, 0, 0L),
+        var unorderedStats = List.of(
+            new ThreadPoolStats.Stats("z", 7, 0, 0, 0, 0, 0L),
+            new ThreadPoolStats.Stats("m", 5, 0, 0, 0, 0, 0L),
+            new ThreadPoolStats.Stats("m", -3, 0, 0, 0, 0, 0L),
+            new ThreadPoolStats.Stats("d", 2, 0, 0, 0, 0, 0L),
+            new ThreadPoolStats.Stats("m", 4, 0, 0, 0, 0, 0L),
+            new ThreadPoolStats.Stats("t", 6, 0, 0, 0, 0, 0L),
             new ThreadPoolStats.Stats("a", -1, 0, 0, 0, 0, 0L)
         );
 
-        var copy = new ArrayList<>(stats);
+        var copy = new ArrayList<>(unorderedStats);
         Collections.sort(copy);
 
-        var names = copy.stream().map(ThreadPoolStats.Stats::name).toList();
-        assertThat(names, contains("a", "d", "m", "m", "m", "t", "z"));
+        assertThat(copy.stream().map(ThreadPoolStats.Stats::name).toList(), contains("a", "d", "m", "m", "m", "t", "z"));
+        assertThat(copy.stream().map(ThreadPoolStats.Stats::threads).toList(), contains(-1, 2, -3, 4, 5, 6, 7));
 
-        var threads = copy.stream().map(ThreadPoolStats.Stats::threads).toList();
-        assertThat(threads, contains(-1, -1, 1, 2, 3, -1, -1));
+        // assert that the ThreadPoolStats constructor sorts the stat list
+        var threadPoolStats = new ThreadPoolStats(unorderedStats);
+        assertThat(threadPoolStats.stats().stream().map(ThreadPoolStats.Stats::name).toList(), contains("a", "d", "m", "m", "m", "t", "z"));
+        assertThat(threadPoolStats.stats().stream().map(ThreadPoolStats.Stats::threads).toList(), contains(-1, 2, -3, 4, 5, 6, 7));
+    }
+
+    public void testMergeThreadPoolStats() {
+        var firstStats = List.of(randomStats("name-1"), randomStats("name-2"), randomStats("name-3"));
+        var secondStats = List.of(randomStats("name-4"), randomStats("name-5"), randomStats("name-2"), randomStats("name-3"));
+
+        var tps1 = new ThreadPoolStats(firstStats);
+        var tps2 = new ThreadPoolStats(secondStats);
+        var target = ThreadPoolStats.merge(tps1, tps2);
+
+        assertThat(target.stats(), hasSize(5));
+        assertThat(
+            target.stats(),
+            containsInAnyOrder(
+                firstStats.get(0), // name-1
+                ThreadPoolStats.Stats.merge(firstStats.get(1), secondStats.get(2)), // name-2
+                ThreadPoolStats.Stats.merge(firstStats.get(2), secondStats.get(3)), // name-3
```

Comment:
yeah, I thought the same while writing this. That's the reason why I wrote a separate test for this method at `testStatsMerge()` (a couple of lines below). IMO, that would give us enough confidence in case a bug is introduced to the other method. WDYT?